src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedSimpleProgramEntryPointSymbol.cs

```diff
@@ -281,6 +281,31 @@ internal override bool IsDefinedInSourceTree(SyntaxTree tree, TextSpan? definedW
 
         public SyntaxNode ReturnTypeSyntax => CompilationUnit.Members.First(m => m.Kind() == SyntaxKind.GlobalStatement);
 
+        /// <summary>
+        /// Returns a bag of applied custom attributes and data decoded from well-known attributes. Returns null if there are no attributes applied on the symbol.
+        /// </summary>
+        /// <remarks>
+        /// Forces binding and decoding of attributes.
+        /// </remarks>
+        internal override CustomAttributesBag<CSharpAttributeData> GetAttributesBag()
+        {
+            if (_lazyCustomAttributesBag == null || !_lazyCustomAttributesBag.IsSealed)
+            {
+                var mergedAttributes = ((SourceAssemblySymbol)this.ContainingAssembly).GetAttributeDeclarations();
+                if (LoadAndValidateAttributes(OneOrMany.Create(mergedAttributes), ref _lazyCustomAttributesBag))
+                {
+                    NoteAttributesComplete(forReturnType: false);
+                }
+            }
+
+            // PROTOTYPE: nullability
+            return _lazyCustomAttributesBag!;
+        }
+
+        AttributeLocation IAttributeTargetSymbol.DefaultAttributeLocation => AttributeLocation.Main;
```

Comment:
>AttributeLocation.Main [](http://example.com/codeflow?start=77&length=22)

This implementation feels suspicious. We are supposed to return "Location of an attribute if an explicit location is not specified via attribute target specification syntax." However, attributes without explicit "main" target shouldn't be applicable to this symbol. Perhaps the default location should be ```AttributeLocation.Assembly```. That is what we use for attributes on assembly?